src/qibolab/result.py

```diff
@@ -6,34 +6,77 @@
 import numpy.typing as npt
 from scipy import signal
 
+ExecRes = np.dtype([("i", np.float64), ("q", np.float64)])
+
 
 @dataclass
-class ExecutionResults:
-    """Data structure to deal with the output of execute_pulse_sequence"""
+class AveragedResults:
+    """Data structure containing averages of ``ExecutionResults``."""
 
     i: npt.NDArray[np.float64]
     q: npt.NDArray[np.float64]
+
+    def __add__(self, data):
+        i = np.append(self.i, data.i)
+        q = np.append(self.q, data.q)
+
+        new_execution_results = self.__class__(i, q)
+
+        return new_execution_results
+
+    def to_dict(self):
+        """Serialize output in dict"""
+
+        return {
+            "MSR[V]": np.sqrt(self.i**2 + self.q**2),
+            "i[V]": self.i,
+            "q[V]": self.q,
+            "phase[rad]": np.angle(self.i + 1.0j * self.q),
+        }
+
+
+@dataclass
+class ExecutionResults:
+    """Data structure to deal with the output of :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`"""
+
+    array: npt.NDArray[ExecRes]
     shots: Optional[npt.NDArray[np.uint32]] = None
 
     @classmethod
     def from_components(cls, is_, qs_, shots=None):
-        return cls(is_, qs_, shots)
+        ar = np.empty(is_.shape, dtype=ExecRes)
+        ar["i"] = is_
+        ar["q"] = qs_
+        ar = np.rec.array(ar)
+        return cls(ar, shots)
 
     @property
-    def in_progress(self):
-        """Placeholder for when we implement live fetching of data from instruments."""
-        return False
+    def i(self):
+        return self.array.i
+
+    @property
+    def q(self):
+        return self.array.q
+
+    def __add__(self, data):
+        i = np.append(self.i, data.i, axis=0)
+        q = np.append(self.q, data.q, axis=0)
+
+        new_execution_results = self.__class__.from_components(i, q)
+
+        return new_execution_results
 
     @cached_property
     def measurement(self):
-        """Resonator signal voltage mesurement (MSR) in volts."""
+        """Computes measurement value."""
```

Comment:
Why did you suppress the `resonator signal voltage`? Assuming it was accurate, it was useful.

In particular, a property *is* something, and not does something, so the former docstring had a better style.

Reply:
Thanks for catching this. This happened probably after merging main.